Yoast/wordpress-seo
# Restore variable_warning and use onInput.

## Summary

This PR can be summarized in the following changelog entry:

* Restores the missing warning for wrong replacement variables.

## Relevant technical choices:

* the translation string `variable_warning` used to be localised in the `wpseoAdminL10n` script
* after the Help Center refactoring `wpseoAdminL10n` is now enqueued only in the post and term pages
* this PR restores the missing string putting it in the `wpseoAdminGlobalL10n` localized script
* also, changes the `change` that triggered the check to `input` event:
  - modern browsers support the `input` event 
  - this way, the warning is displayed as soon as users finish typing a wrong variable, and also on paste
  - previously, the warning was displayed only when the field was blurred 

## Test instructions

This PR can be tested by following these steps:

- on current trunk, go in Search Appearance > Taxonomies > Categories
- enter `%%searchphrase%%` in the Categories Meta description template
- observe the JS error in your browser's console and no warning appearing
- switch to this branch, build the JS
- repeat the steps above
- check the warning appears correctly and no JS errors occur


Note 1: the warning copy could be improved

Note 2: the warning doesn't appear on page load (same also before these changes). Say users enter a wrong variable and submit anyway. After the page reload, or at a subsequent visit of the same page, no warning is displayed. Maybe this could be improved a bit.

Fixes #9866
